hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java

```diff
@@ -210,13 +210,33 @@ protected void incrementCounter(Statistic objectsRead) {
   }
 
   @Override
-  public OzoneFSOutputStream createFile(String key, boolean overWrite,
+  public OzoneFSOutputStream createFile(String key,
+      short replication,
+      boolean overWrite,
       boolean recursive) throws IOException {
     incrementCounter(Statistic.OBJECTS_CREATED);
     try {
-      OzoneOutputStream ozoneOutputStream = bucket
-          .createFile(key, 0, replicationType, replicationFactor, overWrite,
-              recursive);
+      OzoneOutputStream ozoneOutputStream = null;
+      if (replication == ReplicationFactor.ONE.getValue()
+          || replication == ReplicationFactor.THREE.getValue()) {
+        ReplicationFactor clientReplication
+            = ReplicationFactor.valueOf(replication);
+        ozoneOutputStream = bucket.createFile(
+            key,
+            0,
+            replicationType,
+            clientReplication,
+            overWrite,
+            recursive);
+      } else {
+        ozoneOutputStream = bucket.createFile(
+            key,
+            0,
+            replicationType,
+            replicationFactor,
```

Comment:
If the replicationFactor is 2, will the bucket create the key using system default replication here?

Reply:
Yes, if the replicationFactor is not 1 or 3, the default value is used.